### PR TITLE
feat: add a new custom snackbar

### PR DESCRIPTION
**Pull request by UncleBill**
## Usage

```diff
diff --git a/packages/maskbook/src/components/InjectedComponents/ToolboxHint.tsx b/packages/maskbook/src/components/InjectedComponents/ToolboxHint.tsx
index 0ad7b9c17..de27c8fc9 100644
--- a/packages/maskbook/src/components/InjectedComponents/ToolboxHint.tsx
+++ b/packages/maskbook/src/components/InjectedComponents/ToolboxHint.tsx
@@ -1,5 +1,5 @@
 import { CircularProgress, MenuItem, MenuItemProps, Typography } from '@material-ui/core'
-import { makeStyles } from '@masknet/theme'
+import { makeStyles, useSnackbar, CustomSnackbarContentProps, CustomSnackbarContent } from '@masknet/theme'
 import classNames from 'classnames'
 import {
     useAccount,
@@ -246,6 +246,28 @@ export function ToolboxHint(props: ToolboxHintProps) {
 
     const isWalletValid = !!account && selectedWallet && chainIdValid
 
+    const snackbar = useSnackbar()
+
+    const snackbarKeyRef = useRef<string | number>()
+    const showSnackbar = useCallback(
+        (options: Partial<CustomSnackbarContentProps & { persist: boolean }>) => {
+            if (snackbarKeyRef.current) {
+                snackbar.closeSnackbar(snackbarKeyRef.current)
+            }
+            snackbarKeyRef.current = snackbar.enqueueSnackbar('Creating', {
+                variant: options.variant,
+                persist: options.persist ?? true,
+                content: (key, title) => {
+                    return <CustomSnackbarContent id={key} title={title} {...options} />
+                },
+            })
+            return () => {
+                snackbar.closeSnackbar(snackbarKeyRef.current)
+            }
+        },
+        [snackbar],
+    )
+
     function renderButtonText() {
         if (!account) return t('plugin_wallet_on_connect')
         if (!chainIdValid) return t('plugin_wallet_wrong_network')
@@ -264,7 +286,14 @@ export function ToolboxHint(props: ToolboxHintProps) {
 
     return (
         <>
-            <div className={classes.wrapper} onClick={openMenu}>
+            <div
+                className={classes.wrapper}
+                onClick={() => {
+                    showSnackbar({
+                        processing: true,
+                        message: 'Processing',
+                    })
+                }}>
                 <div className={classes.button}>
                     <MaskbookSharpIconOfSize classes={{ root: classes.icon }} size={22} />
                     <Typography className={classes.title}>Mask Network</Typography>

```

## Result

image

### CODE REVIEW

File: packages/theme/package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "start": "dev -- start-storybook -p 6006"
   },
   "dependencies": {
+    "@masknet/icons": "workspace:*",
```
Comment (Jack-Works) [inline packages/theme/package.json:24]: Please also add this package as a project reference in tsconfig.

**Review by NikoManiac — APPROVED** [review]

File: packages/theme/src/Components/Snackbar/index.tsx [hunk 2]
```diff
@@ -1,37 +1,184 @@
...
+                color: 'inherit',
+            },
+        },
+        error: {
+            background: MaskColorVar.redMain,
+            color: MaskColorVar.lightestBackground,
+            '& $title': {
+                color: 'inherit',
+            },
+            '& $message': {
```
Comment (Jack-Works) [inline packages/theme/src/Components/Snackbar/index.tsx:54]: https://github.com/garronej/tss-react/#composition-and-nested-selectors--syntax-

File: packages/theme/package.json (hunk 1 above)
Comment (UncleBill) [reply packages/theme/package.json:24]: Fixed. Thanks.

File: packages/icons/general/Risk.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { createIcon } from '../utils'
+export const RiskIcon = createIcon(
+    'Risk',
+    <>
```
Comment (guanbinrui) [inline packages/icons/general/Risk.tsx:4]:
```suggestion
    <g>
```

File: packages/theme/src/Components/Snackbar/index.tsx (hunk 2 above)
Comment (UncleBill) [reply packages/theme/src/Components/Snackbar/index.tsx:54]: TBH, I think `css-in-js` is painful. Hard to read and write. 😂 I personally prefer `css-modules`. Could we have a try at it?

Comment (Jack-Works) [reply packages/theme/src/Components/Snackbar/index.tsx:54]: We need to take control of when the styles are injected therefore we can make sure they work in content scripts. If you think nested selectors are not easy to use in `makeStyles`, you can try this:

```ts
const Type1 = styled("div")`
    some css
`
const Type2 = styled("div")`
    ${Type1} & {
        some other styles.
    }
`
```

**Comment by UncleBill** [comment]
Link https://github.com/DimensionDev/Maskbook/pull/3718/files

File: packages/icons/general/Risk.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { createIcon } from '../utils'
+export const RiskIcon = createIcon(
+    'Risk',
+    <g>
+        <path
```
Comment (septs) [inline packages/icons/general/Risk.tsx:5]: plz use https://aka.pw/svgomg reduce svg image size

**Pull request merged**